### ISSUE TITLE
Arbitrary Style Transfer in Real-time with Adaptive Instance Normalization

### ISSUE THREAD

**Issue by HYUNMIN-HWANG** [yskim0/One-Day-One-Paper#23]
# Arbitrary Style Transfer in Real-time with Adaptive Instance Normalization
- https://arxiv.org/abs/1703.06868
- code : https://github.com/xunhuang1995/AdaIN-style
- [Submitted on 20 Mar 2017 (v1), last revised 30 Jul 2017 (this version, v2)]
- ICCV 2017

---

## Summary
기존  style transfer 들은 속도가 느리거나 / 한 가지 스타일로만 변환할 수 있다는 단점이 있다. 이를 극복하기 위해서 해당 논문에서는 Adaptive Instance Normalization를 사용해 한 번 훈련시키면 임의의 스타일로 변환할 수 있는 style transfer 모델을 제안했다.

## Normalization
1. Batch Normalization
![image](https://user-images.githubusercontent.com/70581043/128135897-54c25154-8dde-4d99-9d4e-c814371195a4.png)
네트워크를 빠르게 훈련시킬 수 있다는 장점이 있다.  each feature channel 에서의 mean 값과 standard deviation을 계산한다.

2. Instance Normalization
![image](https://user-images.githubusercontent.com/70581043/128136030-65c7877d-71c8-405e-9bf4-92852f2520f2.png)
Batch Normalization와 유사함, Instance Normalization은 each feature channel 뿐만 아니라 each sample에서의 mean 값과 standard deviation을 계산한다.

3. Adaptive Instance Normalization
![image](https://user-images.githubusercontent.com/70581043/128136185-2f007687-c251-4e50-86b6-57400d91cd77.png)
각각의 스타일 s 마다 다른 감마, 베타 값을 다르게 적용시킬 수 있다. 즉, 하나의 같은 네트워크에서 완전히 다른 스타일의 이미지를 만들어낼 수 있다. *(논문에서 참고한 방식)*

4. Adaptive Instance Normalization (AdaIN)
![image](https://user-images.githubusercontent.com/70581043/128136593-a8f77fbf-dec4-4011-b1c1-5c219208084b.png)
content 를 전달할 x 이미지를 넣었을 때, y의 style 로 바꿔주는 역할을 한다. *(논문에서 제안한 방식)*

## Architecture
![image](https://user-images.githubusercontent.com/70581043/128136897-39cf9fe2-b792-438f-a158-37a83885272b.png)

- encoder f :  pre-trained VGG-19 , content image와 style image에서의 feature를 추출한다.
- AdaIN layer : content feature map에서 style feature map으로 변환해준다.
- decoder g : 스타일이 적용된 이미지를 generate 한다.
- pre-trained VGG-19 : content loss 와 style loss 계산

> - content loss
![image](https://user-images.githubusercontent.com/70581043/128137936-7e7e2e66-fa14-439d-b3bb-76101aaa4447.png)
아웃풋 이미지의 features와 타겟 features의 거리 차이를 구한다.

> - style loss
![image](https://user-images.githubusercontent.com/70581043/128137966-28950b12-7838-4fd0-86db-eac463014027.png)
IN statistics 를 사용하여 결과를 계산한다. 각각의 VGG19 레이어에서의 style loss를 구한다.

## Result
![image](https://user-images.githubusercontent.com/70581043/128137473-5d875c16-3e85-4a3e-80de-43e01308901e.png)
 - content loss 와 style loss 가 어느 정도 좋은 결과를 보임
![image](https://user-images.githubusercontent.com/70581043/128137558-2ea36b7c-34d1-4100-8508-624560883dc2.png)
- 기존 모델에 비해 속도가 빠른 편이며, 다양한 스타일로 변환할 수 있다.
![image](https://user-images.githubusercontent.com/70581043/128137621-f1e43e63-fcd5-4243-8f5e-1a7771e7f973.png)
- 눈으로 봤을 때도 스타일이 잘 적용된 것을 볼 수 있다.
